app/database/operator/server_data_operator/handlers/channel.ts

```diff
@@ -246,10 +246,12 @@ const ChannelHandler = <TBase extends Constructor<ServerDataOperatorBase>>(super
                 const totalMsg = isCRT ? channel.total_msg_count_root! : channel.total_msg_count;
                 const myMsgCount = isCRT ? my.msg_count_root! : my.msg_count;
                 const msgCount = Math.max(0, totalMsg - myMsgCount);
+                const lastPostTimestamp = isCRT ? (channel.last_root_post_at || channel.last_post_at) : channel.last_post_at;
+                const lastPostAt = Math.max(lastPostTimestamp, channel.create_at);
                 my.msg_count = msgCount;
                 my.mention_count = isCRT ? my.mention_count_root! : my.mention_count;
                 my.is_unread = msgCount > 0;
-                my.last_post_at = (isCRT ? (channel.last_root_post_at || channel.last_post_at) : channel.last_post_at) || 0;
+                my.last_post_at = lastPostAt;
```

Comment:
you are right, the channel does show.. let me fix that.

Reply:
@larkox is it only for DM's ? what about GM and other channel types?

Reply:
As far as I know... DMs for sure. GMs I am not sure, that would be something to ask UX I would say. They have the same "problem" as DMs (you may create one "by mistake" and then never send a message), but they are a bit different.

Other channels (private and public) should show, since the membership is never made by mistake. You either joined actively the channel, or where actively added to the channel.

Reply:
Oh yeah, I forgot a piece of logic in the web app. Not only do we always show a channel if it's unread, we also show it if it's the current channel. I guess that might work differently for mobile though because you're not immediately viewing a channel after you've created it

Other users only see DMs/GMs once someone posts in that channel